ignite/templates/typed/singleton/singleton.go

```diff
@@ -96,58 +98,59 @@ const (
 	}
 }
 
-func protoRPCModify(replacer placeholder.Replacer, opts *typed.Options) genny.RunFn {
+// Modifies query.proto to add the required RPCs and Messages.
+//
+// What it depends on:
+//   - Existence of a service with name "Query". Adds the rpc's there.
+func protoRPCModify(opts *typed.Options) genny.RunFn {
 	return func(r *genny.Runner) error {
-		path := filepath.Join(opts.AppPath, "proto", opts.AppName, opts.ModuleName, "query.proto")
+		path := opts.ProtoPath("query.proto")
 		f, err := r.Disk.Find(path)
 		if err != nil {
 			return err
 		}
+		pf, err := protoutil.ParseProtoFile(f)
+		if err != nil {
+			return err
+		}
 
-		// Import the type
-		templateImport := `import "%s/%s/%s.proto";
-%s`
-		replacementImport := fmt.Sprintf(templateImport,
-			opts.AppName,
-			opts.ModuleName,
-			opts.TypeName.Snake,
-			typed.Placeholder,
-		)
-		content := replacer.Replace(f.String(), typed.Placeholder, replacementImport)
-
-		// Add gogo.proto
-		replacementGogoImport := typed.EnsureGogoProtoImported(path, typed.Placeholder)
-		content = replacer.Replace(content, typed.Placeholder, replacementGogoImport)
-
-		// Add the service
-		templateService := `// Queries a %[2]v by index.
-	rpc %[2]v(QueryGet%[2]vRequest) returns (QueryGet%[2]vResponse) {
-		option (google.api.http).get = "/%[3]v/%[4]v/%[5]v";
-	}
-%[1]v`
+		// Import the type and gogoproto.
+		gogoproto := protoutil.NewImport("gogoproto/gogo.proto")
+		if err = protoutil.AddImports(pf, true, gogoproto, opts.TypeImport()); err != nil {
+			return fmt.Errorf("failed while adding imports in %s: %w", path, err)
+		}
+		// Find service.
+		srv, err := protoutil.GetServiceByName(pf, "Query")
```

Comment:
should I also go around and think of better names for all these? I generally use shorthand versions for many things but I could give them longer descriptive names.

Reply:
@aljo242  Considering your previous review [here](https://github.com/ignite/cli/pull/3109#discussion_r1032724039) I thought about names used in the functions that generate the protos. I commonly use shorthand names like `typU` for `typename.Upper` or `typS` for `typename.Snake`, etc, and was wondering if those should also be renamed to something more descriptive.

Reply:
yeah: things like `typU -> `typenameUpper` etc.